src/Components/ShortcutKeyModal/SearchModal.js

```diff
@@ -0,0 +1,142 @@
+import React, { useRef, useEffect, useState, useCallback, useMemo } from 'react';
+import './SearchModal.css';
+import { officerOrAdminRoutes, signedOutRoutes, memberRoutes, notAuthenticatedRoutes } from '../../Routes';
+import { membershipState } from '../../Enums';
+import { useUser } from '../context/UserContext';
+
+export default function SearchModal({ appProps }) {
+  const [open, setOpen] = useState(false);
+  const inputRef = useRef(null);
+  const [keyword, setKeyword] = useState('');
+  const [suggestions, setSuggestions] = useState([]);
+  const [selectItem, setSelectItem] = useState(0);
+  const { user } = useUser();
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const routes = useMemo(() => {
+    if (user.accessLevel === membershipState.MEMBER) return [...memberRoutes, ...signedOutRoutes];
+    if (user.accessLevel >= membershipState.OFFICER) return [...officerOrAdminRoutes, ...signedOutRoutes];
+    if (!appProps.authenticated) return [...notAuthenticatedRoutes, ...signedOutRoutes];
+    return [...signedOutRoutes];
+  }, [user.accessLevel]);
+
+  function handleChanges(e) {
+    setKeyword(e.target.value);
+    setSelectItem(0);
+  }
+
+  function getSuggestions() {
+    if (suggestions.length === 0) return <></>;
+
+    return (
+      <ul className='suggestion-list'>
+        {suggestions.map((r, index) => (
```

Comment:
lets move the slice up here, easier to read
```suggestion
    const firstFiveSuggestions = suggestions.slice(0, 5);
    return (
      <ul className='suggestion-list'>
        {firstFiveSuggestions.map((r, index) => (
```

Reply:
Done